mentat/session_input.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+import logging
+from typing import Any, Coroutine
+
+from .errors import RemoteKeyboardInterrupt
+from .session_stream import StreamMessage, get_session_stream
+
+logger = logging.getLogger()
+
+
+async def collect_user_input() -> StreamMessage:
+    """Listens for user input on a new channel
+
+    send a message requesting user to send a response
+    create a new broadcast channel that listens for the input
+    close the channel after receiving the input
+    """
+    stream = get_session_stream()
+
+    message = await stream.send("", channel="input_request")
+    response = await stream.recv(f"input_request:{message.id}")
+
+    return response
+
+
+async def ask_yes_no(default_yes: bool) -> bool:
+    stream = get_session_stream()
+
+    while True:
+        # TODO: combine this into a single message (include content)
+        await stream.send("(Y/n)" if default_yes else "(y/N)")
+        response = await collect_user_input()
+        content = response.data
+        if content in ["y", "n", ""]:
+            break
+    return content == "y" or (content != "n" and default_yes)
+
+
+async def listen_for_interrupt(
```

Comment:
This isn't really used right now (it will be used once we drop `StreamingPrinter`)